testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/providers/jackson2/whitelist/WhiteListPolymorphicTypeValidatorManualOverrideTest.java

```diff
@@ -83,7 +83,7 @@ public void testAircraftFailure() throws Exception {
         String response = sendPost(new TestPolymorphicType(new Aircraft()));
         logger.info("response: " + response);
         Assert.assertNotNull(response);
-        Assert.assertTrue(response.contains("Response code: " + HttpResponseCodes.SC_BAD_REQUEST));
+        Assert.assertTrue(response.contains("Response code: " + HttpResponseCodes.SC_INTERNAL_SERVER_ERROR));
```

Comment:
This seems kind of debatable. If a request entity fails the whitelist, isn't that a client problem?

Reply:
I don't fully disagree with you. The only reason I did this was because the spec suggests to return 500. While that isn't required, I think it's best to do what the spec suggests there. However, I do agree that is up for debate and really the spec should state a "MUST" or not state it at all IMO.